internal/storage/v1/elasticsearch/factory.go

```diff
@@ -41,12 +42,20 @@ const (
 )
 
 var ( // interface comformance checks
-	_ storage.Factory        = (*Factory)(nil)
-	_ io.Closer              = (*Factory)(nil)
-	_ storage.Configurable   = (*Factory)(nil)
-	_ storage.Inheritable    = (*Factory)(nil)
-	_ storage.Purger         = (*Factory)(nil)
-	_ storage.ArchiveCapable = (*Factory)(nil)
+	_                storage.Factory        = (*Factory)(nil)
+	_                io.Closer              = (*Factory)(nil)
+	_                storage.Configurable   = (*Factory)(nil)
+	_                storage.Inheritable    = (*Factory)(nil)
+	_                storage.Purger         = (*Factory)(nil)
+	_                storage.ArchiveCapable = (*Factory)(nil)
+	disableLegacyIDs                        = featuregate.GlobalRegistry().MustRegister(
+		"jaeger.es.disableLegacyId",
+		featuregate.StageBeta, // enabed by default
+		featuregate.WithRegisterFromVersion("v2.5.0"),
+		featuregate.WithRegisterToVersion("v2.8.0"),
+		featuregate.WithRegisterDescription("Legacy trace ids are the ids having starting 0s, enabling this will force the reader to search for the spans with trace ids having leading zeroes"),
+		featuregate.WithRegisterReferenceURL("https://github.com/jaegertracing/jaeger/issues/1578"),
+	)
```

Comment:
We need to add the handling of feature flags via CLI to v1 components, right now it's not possible to change the value except in jaeger-v2.

Reply:
see #6790 for example